modules/stochastic_tools/src/userobjects/NeuralNetworkUserObject.C

```diff
@@ -0,0 +1,193 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NeuralNetworkUserObject.h"
+#include "MooseEnum.h"
+#include "pugixml.h"
+
+registerMooseObject("StochasticToolsApp", NeuralNetworkUserObject);
+
+template <>
+InputParameters
+validParams<NeuralNetworkUserObject>()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params.addClassDescription("Reconstructs a neural network from a file and evaluates it");
+  params.addRequiredParam<FileName>("weights_file", "Name of the file with the neuron weights");
+  return params;
+}
+
+NeuralNetworkUserObject::NeuralNetworkUserObject(const InputParameters & parameters)
+  : GeneralUserObject(parameters),
+    _weightsFile(getParam<FileName>("weights_file")),
+    _layerActivationFunctionEnum(MultiMooseEnum("SIGMOID SOFTSIGN TANH LINEAR LOGSIGMOID" ) )
+{
+  setXMLWeights();
+
+}
+void NeuralNetworkUserObject::setXMLWeights()
+{
+  std::ifstream ifile;
+  std::string file_dump;
+  pugi::xml_document doc;
+
+  ifile.open(_weightsFile);
+  if (!ifile)
+    {
+      paramError("Cannot read neural net file");
+    }
+  pugi::xml_parse_result result = doc.load(ifile);
```

Comment:
This is likely going away, we are moving to JSON format

Reply:
@aeslaughter in the interest of not letting this PR die once Chaitanya's internship ends (pretty soon), I suggest going with XML in this PR and transitioning over to JSON in a separate PR. I have interest in that as well and have some code that I will combine with Chaitanya's stuff.

Reply:
That is fine. The biggest thing I would like done then is using python/stochastic_tools and at least have a few tests to make sure these tools run.